transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/TransportRuntime.java

```diff
@@ -83,6 +84,12 @@ public static TransportRuntime getInstance() {
     return localRef.getTransportRuntime();
   }
 
+  @VisibleForTesting
+  @RestrictTo(RestrictTo.Scope.TESTS)
+  static void setInstance(TransportRuntimeComponent component) {
```

Comment:
I'd suggest you replace this with a higher-order function like:

```java
static void withInstance(TransportRuntimeComponent component, Runnable runnable) {
  TransportRuntimeComponent original;
  synchronized (TransportRuntime.class) {
      original = INSTANCE;
      INSTANCE = component;
  }
  try {
    runnable.run();
  } finally {
    synchronized (TransportRuntime.class) {
      INSTANCE = original;
    }
  }

}
```

This will make sure that, no matter what failure happens, the runtime is restored to its original state. Otherwise there's big potential for cascading test failures that run after such a failed test.